src/sentry/sentry_metrics/indexer/cloudspanner.py

```diff
@@ -13,16 +13,18 @@
 
 class IdCodec(Codec[DecodedId, EncodedId]):
     """
-    Reverses 64 bit IDs for storage. IDs generated by our id_generator are incrementing and prefixed
-    with a fixed version but we need well distributed primary keys to spread the work evenly
-    between CloudSpanner nodes.
+    Encodes 63 bit IDs generated by the id_generator so that they are well distributed for CloudSpanner.
+
+    Given an ID, this codec does the following:
+    - reverses the bits and shifts to the left by one
+    - Subtract 2^63 so that that the unsigned 64 bit integer now fits in a signed 64 bit field
     """
 
     def encode(self, value: DecodedId) -> EncodedId:
-        return reverse_bits(value, 64)
+        return reverse_bits(value, 64) - 2**63
```

Comment:
Nit
```suggestion
        return reverse_bits(value, 64) - (sys.maxsize + 1)
```

Reply:
@nikhars it probably makes sense to have a constant for this, but sys.maxsize defines how large a pointer in the current python process is. so hypothetically if this python process runs on a 32-bit machine, this value would be 32. I don't think we want that, because we're talking about db integer width, right?

btw, afaik this value is not about how large an int in python can be. those are arbitrary width

Reply:
Yep. sys.maxsize depends on the platform it's being run on which we don't want.